packages/core/echo/echo-db/src/query/query.test.ts

```diff
@@ -207,12 +205,53 @@ describe('Queries', () => {
     {
       const peer = await builder.createPeer(kv);
       const db = await peer.openDatabase(spaceKey, root);
-      const queryResult = (await db.query().run({ timeout: 200 })).objects;
+      const queryResult = (await db.query().run()).objects;
       expect(queryResult.length).to.eq(1);
       expect(queryResult[0].id).to.eq(expectedObjectId);
     }
   });
 
+  test('objects url changes, the latest document is loaded', async () => {
+    const kv = createTestLevel();
+    const spaceKey = PublicKey.random();
+    const builder = new EchoTestBuilder();
+    afterTest(() => builder.close());
+
+    let root: AutomergeUrl;
+    let assertion: { objectId: string; documentUrl: string };
+    {
+      const peer = await builder.createPeer(kv);
+
+      const db = await peer.createDatabase(spaceKey);
+      const [obj1, obj2] = await createObjects(peer, db, { count: 2 });
+
+      expect((await db.query().run()).objects.length).to.eq(2);
+      const rootDocHandle = db.coreDatabase._automergeDocLoader.getSpaceRootDocHandle();
+      const anotherDocHandle = getObjectCore(obj2).docHandle!;
+      anotherDocHandle.change((doc: SpaceDoc) => {
+        doc.objects![obj1.id] = getObjectCore(obj1).docHandle!.docSync().objects![obj1.id];
+      });
+      rootDocHandle.change((doc: SpaceDoc) => {
+        doc.links![obj1.id] = anotherDocHandle.url;
+      });
+      await db.flush();
+      await peer.host.queryService.reindex();
+
+      root = rootDocHandle.url;
+      assertion = { objectId: obj2.id, documentUrl: anotherDocHandle.url };
+    }
+
+    {
```

Comment:
there's a better `peer.reload()` api which similates a restart without you having to pass `kv` around